dag_in_context/src/greedy_dag_extractor.rs

```diff
@@ -354,30 +355,59 @@ fn calculate_cost_set(
 
     let mut shared_total = NotNan::new(0.).unwrap();
     let mut unshared_total = info.cm.get_op_cost(&node.op);
-    let mut costs: HashMap<ClassId, NotNan<f64>> = HashMap::default();
+    let mut costs: HashTrieMap<ClassId, NotNan<f64>> = Default::default();
```

Comment:
Do you have a sense of how persistent data structures improve performance? Functional data structures in general have a much larger constant factor.

Reply:
We get more sharing in memory- we didn't get much of a speedup as far as I can tell, but memory usage is halved

Reply:
Actually perhaps we did get a big speedup- looking at the numbers below